api/users/serializers.py

```diff
@@ -195,6 +198,16 @@ class ReadEmailUserDetailSerializer(UserDetailSerializer):
     email = ser.CharField(source='username', read_only=True)
 
 
+class UserMeDetailSerializer(ReadEmailUserDetailSerializer):
```

Comment:
Inheriting from `ReadEmailUserDetailSerializer` seems wrong -- `email` will be included even if the token is missing `USER_EMAIL_READ` scope. Better and [more consistent](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/base/serializers.py#L81) would be a custom `ShowIfCurrentUser` field that overrides `get_attribute` and raises `SkipField`

Reply:
👍